app/controllers/schools/choose_programme_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Schools::ChooseProgrammeController < Schools::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  def show
+    school = @current_user.induction_coordinator_profile.schools.first
+    redirect_to helpers.profile_dashboard_url(@current_user) if school.chosen_programme?(Cohort.current)
```

Comment:
I think this should be redirect_to and return

Reply:
I went with `if-else` statement, but you are right, anything that happens after `redirect_to` will still be fired off :/